components/Registration/Registration.tsx

```diff
@@ -86,14 +90,170 @@ const RegistrationForm: React.FC = () => {
         handlePageChange(formIndex - 1);
     };
 
-    const onSubmit = (
+    const createRegistrationObject = async (
+        fieldValues: FieldValues
+    ): Promise<RegistrationType> => {
+        const registrationWithId = await getRegistration();
```

Comment:
Can this GET request be done only on first page load and then the registration object be kept up to date as the user fills in the form?